c/authService.c

```diff
@@ -173,6 +174,202 @@ static int serveAuthCheck(HttpService *service, HttpResponse *res) {
   return 0;
 }
 
+int resetPassword(HttpService *service, HttpResponse *response) {
+  int returnCode = 0, reasonCode = 0;
+  jsonPrinter *p = respondWithJsonPrinter(response);
+  HttpRequest *request = response->request;
+
+  if (!strcmp(request->method, methodPOST)) {
+    char *inPtr = request->contentBody;
+    char *nativeBody = copyStringToNative(request->slh, inPtr, strlen(inPtr));
+    int inLen = nativeBody == NULL ? 0 : strlen(nativeBody);
+    char errBuf[JSON_ERROR_BUFFER_SIZE];
+
+    Json *body = jsonParseUnterminatedString(request->slh, nativeBody, inLen, errBuf, JSON_ERROR_BUFFER_SIZE);
+    JsonObject *inputMessage = jsonAsObject(body);
+    Json *username = jsonObjectGetPropertyValue(inputMessage,"username");
+    Json *password = jsonObjectGetPropertyValue(inputMessage,"password");
+    Json *newPassword = jsonObjectGetPropertyValue(inputMessage,"newPassword");
+    int usernameLength = strlen(jsonAsString(username));
+    int passwordLength = strlen(jsonAsString(password));
+    int newPasswordLength = strlen(jsonAsString(newPassword));
+
+    if (usernameLength == 0) {
+      setResponseStatus(response, HTTP_STATUS_BAD_REQUEST, "ERROR");
+      setContentType(response,"text/json");
+      addStringHeader(response,"Server","jdmfws");
+      addStringHeader(response,"Transfer-Encoding","chunked");
+      addStringHeader(response, "Cache-control", "no-store");
+      addStringHeader(response, "Pragma", "no-cache");
+      writeHeader(response);
+
+      jsonStart(p);
+      jsonAddString(p, "status", "No username provided");
+      jsonEnd(p);
+
+      finishResponse(response);
+      return HTTP_SERVICE_FAILED;
+    } else if (passwordLength == 0) {
+      setResponseStatus(response, HTTP_STATUS_BAD_REQUEST, "ERROR");
+      setContentType(response,"text/json");
+      addStringHeader(response,"Server","jdmfws");
+      addStringHeader(response,"Transfer-Encoding","chunked");
+      addStringHeader(response, "Cache-control", "no-store");
+      addStringHeader(response, "Pragma", "no-cache");
+      writeHeader(response);
+
+      jsonStart(p);
+      jsonAddString(p, "status", "No password provided");
+      jsonEnd(p);
+
+      finishResponse(response);
+      return HTTP_SERVICE_FAILED;
+	} else if (newPasswordLength == 0) {
+      setResponseStatus(response, HTTP_STATUS_BAD_REQUEST, "ERROR");
```

Comment:
lack of indentation pains me :cry:

Reply:
its indented for me on notepad++ but when its here its not indented not sure if its github adjusting width

Reply:
Emacs shows:
» }·else·if·(newPasswordLength·==·0)·{$
······setResponseStatus(response,·HTTP_STATUS_BAD_REQUEST,·"ERROR");$

The conditionals use tab, the other lines use spaces. This is apparently confusing github.
I can't figure out how this was done, because only part of the new code has tabs.
There should never be tabs in source code. If you could please clean this up I bet it would show correctly on github.

Reply:
@sakeerthy please don't use tabs, replace your new tabs with spaces (2 spaces per tab) and it should solve this indentation issues